k2/csrc/array_ops.cu

```diff
@@ -1,36 +1,26 @@
 /**
  * @brief
- * ops_inl
- *
- * @note
- * Don't include this file directly; it is included by ops.h.
- * It contains implementation code.
+ * array_ops
  *
  * @copyright
- * Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey)
- *                      Fangjun Kuang (csukuangfj@gmail.com)
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey
+ *                                                   Haoweng Qiu
+ *                                                   Meixu Song)
+ *
  * @copyright
  * See LICENSE for clarification regarding multiple authors
  */
 
-#ifndef IS_IN_K2_CSRC_OPS_H_
-#error "this file is supposed to be included only by ops.h"
-#endif
-
-// No header guard for this file since it will only be included
-// in ops.h
-
-#include <cassert>
-#include <cub/cub.cuh>
-#include <type_traits>
+#include "k2/csrc/array_ops.h"
+#include "k2/csrc/utils.cuh"
 
```

Comment:
should not rename an _inl.h file to .cu, should be .cuh.  This is a header and needs to be b/c of template use.

Reply:
Did not. Changed _inl => _inl.cuh. And add split only-cuda related (e.g. lambda) code into .cu.